packages/core/src/internal/utils/global.ts

```diff
@@ -50,26 +41,41 @@ export function setupCDSGlobal() {
   if (isBrowser()) {
     initializeCDSGlobal();
     setRunningVersion();
+    intializeCDSStateProxy();
   }
 }
 
-function getVersion() {
-  if (isBrowser()) {
-    const log: CDSLog = {
-      versions: window.CDS._version,
-      environment: window.CDS.environment,
-      userAgent: navigator.userAgent,
-      supports: window.CDS._supports,
-      angularVersion: getAngularVersion(false),
-      angularJSVersion: getAngularJSVersion(false),
-      reactVersion: getReactVersion(false),
-      vueVersion: getVueVersion(false),
-      loadedElements: window.CDS._loadedElements,
-    };
-    return log;
+function intializeCDSStateProxy() {
+  if (!window.CDS._stateProxy) {
+    window.CDS._stateProxy = true;
+    window.CDS._state = new Proxy(window.CDS._state, {
+      set: (target: any, key: string, value) => {
+        const detail = { key, prev: (window.CDS._state as any)[key], current: value };
+        target[key] = value;
+        document.dispatchEvent(new CustomEvent('CDS_STATE_UPDATE', { detail }));
+        return true;
+      },
+    });
   }
+}
 
-  return void 0;
+function getVersion() {
```

Comment:
If we are focused here on returning the version info, do we need to include the state too?

Reply:
I was thinking for debugging issues getting a snapshot of the state might help triage. Maybe the name here could be different? Maybe `getDetails()` and `logDetails()`?

Reply:
I think either would be better. thanks!